samples/extensions/descriptor_buffer_basic/descriptor_buffer_basic.h

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2023, Sascha Willems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Using descriptor buffers replacing descriptor sets with VK_EXT_descriptor_buffer
+ * This render multiple cubes passing uniform buffers and combined image samples to the GPU using descriptor buffers instead of descriptor sets
+ * This allows for a more bindless design
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+
+class DescriptorBufferBasic : public ApiVulkanSample
+{
+  public:
+	bool animate = true;
+
+	std::unique_ptr<vkb::core::Buffer> resource_descriptor_buffer;
+	std::unique_ptr<vkb::core::Buffer> image_descriptor_buffer;
+
+	VkPhysicalDeviceDescriptorBufferPropertiesEXT descriptor_buffer_properties{};
+
+	VkDescriptorSetLayout descriptor_set_layout_buffer;
+	VkDescriptorSetLayout descriptor_set_layout_image;
+
+	struct Cube
+	{
+		Texture                            texture;
+		std::unique_ptr<vkb::core::Buffer> uniform_buffer;
+		glm::vec3                          rotation;
+		glm::mat4                          model_mat;
+	};
+	std::array<Cube, 2> cubes;
+
+	struct Models
+	{
+		std::unique_ptr<vkb::sg::SubMesh> cube;
+	} models;
+
+	struct UniformBuffers
+	{
+		std::unique_ptr<vkb::core::Buffer> scene;
+	} uniform_buffers;
+
+	struct UboScene
+	{
+		glm::mat4 projection;
+		glm::mat4 view;
+	} ubo_scene;
+
+	VkPipeline       pipeline;
+	VkPipelineLayout pipeline_layout;
+
+	DescriptorBufferBasic();
+	~DescriptorBufferBasic();
```

Comment:
virtual class requires a virtual destructor.  Please use override here.

Reply:
Thanks for bringing this up. Will be fixed.